lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -132,15 +135,28 @@ export class PublicClientApplication implements IPublicClientApplication {
      * - if true, performs logic to cache and navigate
      * - if false, handles hash string and parses response
      */
-    private async handleRedirectResponse(): Promise<AuthenticationResult> {
+    private async handleRedirectResponse(): Promise<AuthenticationResult | null> {
         // Get current location hash from window or cache.
         const { location: { hash } } = window;
-        const cachedHash = this.browserStorage.getItem(this.browserStorage.generateCacheKey(TemporaryCacheKeys.URL_HASH), CacheSchemaType.TEMPORARY) as string;
         const isResponseHash = UrlString.hashContainsKnownProperties(hash);
-        const loginRequestUrl = this.browserStorage.getItem(this.browserStorage.generateCacheKey(TemporaryCacheKeys.ORIGIN_URI), CacheSchemaType.TEMPORARY) as string;
 
-        const currentUrlNormalized = UrlString.removeHashFromUrl(window.location.href);
+        // Check for interaction type
+        if (isResponseHash) {
+            // Deserialize hash fragment response parameters.
+            const hashUrlString: UrlString = new UrlString(hash);
+            const serverParams: ServerAuthorizationCodeResponse = hashUrlString.getDeserializedHash<ServerAuthorizationCodeResponse>();
+            const requestStateObj: RequestStateObject = ProtocolUtils.parseRequestState(this.browserCrypto, serverParams.state);
+            const platformStateObj: BrowserStateObject = BrowserProtocolUtils.parseBrowserRequestState(this.browserCrypto, requestStateObj.libraryState.platformState);
+            if (StringUtils.isEmpty(platformStateObj.interactionType) || platformStateObj.interactionType !== InteractionType.REDIRECT) {
```

Comment:
I think `platformStateObj.interactionType !== InteractionType.REDIRECT` is the only check that is needed.

Reply:
Added in #2078